test.py

```diff
@@ -55,32 +78,68 @@ async def port_function(transmit_packet_numbers):
     print('port connect')
     my_port.connect()
 
+    # create the transaction config
+    xtion_config = TransactionConfig(
+        src_xtor='raqas',
+        dst_xtors=['enet_xtor_0'],
+        xtor_type='enet_xtor',
+        sent_sql_path='temp/sql.db'
+    )
+
+    # set the transaction config
+    my_port.set_transaction_config(xtion_config)
+
     # create the stream config, set it, and add it to the port
+    stream_id = 0
+    transmit_packet_numbers = [10, 5]
     for number in transmit_packet_numbers:
         stream_config = StreamConfig(
             total_packet_num=number,
             destination_address_hex='FFFFFFFFFFFF',
             source_address_hex='000000000000',
-            frame_byte_num=64
+            frame_byte_num=64,
+            end_stream_action='{}'.format(stream_id)
         )
         my_port.add_stream(stream_config)
 
-    # start sending streams
-    print('start transmit')
-    loop = asyncio.get_running_loop()
-    await loop.run_in_executor(None, my_port.start_transmit)
+    commands_map = {
+        'start_transmit' : start_transmit,
+        'stop_transmit' : stop_transmit,
+        'collect_statistics' : collect_statistics
+    }
+
+    for commands in commands_list:
+        for command in commands:
+            await commands_map[command](my_port)
 
     # disconnect in the end
     print('port disconnect')
-    my_port.disconnect()
+    loop = asyncio.get_running_loop()
+    await loop.run_in_executor(None, my_port.disconnect)
 
-async def main():
-    transmit_packet_numbers = [10, 5]
+async def start_transmit(port):
+    print('start transmit')
+    loop = asyncio.get_running_loop()
+    await loop.run_in_executor(None, port.start_transmit)
+
+async def stop_transmit(port):
+    print('stop transmit')
+    loop = asyncio.get_running_loop()
+    await loop.run_in_executor(None, port.stop_transmit)
 
+async def collect_statistics(port):
+    print('collect statistics')
+    loop = asyncio.get_running_loop()
+    await loop.run_in_executor(None, port.collect_statistics)
+
+async def main():
+    tx_commands = ['start_transmit', 'stop_transmit']
+    rx_commands = ['collect_statistics']
```

Comment:
我覺得你要不要直接給function就好了
```suggestion
    tx_commands = [start_transmit, stop_transmit]
    rx_commands = [collect_statistics]
```

Reply:
No needed